service_test.go

```diff
@@ -66,3 +66,79 @@ func TestIsDisplay(t *testing.T) {
 		}
 	}
 }
+
+func TestFrameBuffer(t *testing.T) {
+	// Make sure that we are using our unit-test version of `exec.Command`.
+	newExecCommand = fakeExecCommand
+
+	t.Run("Default behavior", func(t *testing.T) {
+		frameBuffer, err := NewFrameBuffer()
+		if err != nil {
+			t.Fatalf("Could not create frame buffer: %s", err.Error())
+		}
+		if frameBuffer.Display != "1" {
+			t.Errorf("frameBuffer.Display = %s, want %s", frameBuffer.Display, "1")
+		}
+		args := frameBuffer.cmd.Args[3:]
+		if len(args) != 5 {
+			t.Errorf("args length = %d, want = %d", len(args), 5)
+		} else {
+			if args[0] != "Xvfb" {
+				t.Errorf("args[0] = %s, want = %s", args[0], "Xvfb")
+			}
+			if args[1] != "-displayfd" {
+				t.Errorf("args[1] = %s, want = %s", args[1], "-displayfd")
+			}
+			if args[2] != "3" {
+				t.Errorf("args[2] = %s, want = %s", args[2], "3")
+			}
+			if args[3] != "-nolisten" {
+				t.Errorf("args[3] = %s, want = %s", args[3], "-nolisten")
+			}
+			if args[4] != "tcp" {
+				t.Errorf("args[4] = %s, want = %s", args[4], "tcp")
+			}
+		}
+	})
+	t.Run("With screen size", func(t *testing.T) {
+		options := FrameBufferOptions{
+			ScreenSize: "1024x768x24",
+		}
+		frameBuffer, err := NewFrameBufferWithOptions(options)
+		if err != nil {
+			t.Fatalf("Could not create frame buffer: %s", err.Error())
+		}
+		if frameBuffer.Display != "1" {
+			t.Errorf("frameBuffer.Display = %s, want %s", frameBuffer.Display, "1")
+		}
+		args := frameBuffer.cmd.Args[3:]
```

Comment:
Use cmp.Diff instead of comparing individual elements. It is a lot less code.

https://godoc.org/github.com/google/go-cmp/cmp#Diff

Same above.

Still, this seems like a change detector test and isn't that useful:

https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

I would instead do something like:

1) Start a FrameBuffer with a particular, non-default size and/or depth.
2) Interrogate the display using 

```
d, err := xgbutil.NewConnDisplay(frameBuffer.Display)
if err != nil { ... }
defer d.Conn().Close()
s := d.Screen()
if s.WidthInPixels != 1024 {
 t.Errorf(...)
}
if s.HeightInPixels != 786 {
  t.Errorf(...)
}
...
```

https://godoc.org/github.com/BurntSushi/xgbutil

Reply:
The point of the test is that we don't actually have a frame buffer.  I don't know what your setup is on the Github side, but I would assume that "Xvfb" and friends are not installed.  If you know that they're present and working, I can make an integration test that validates that what we asked for is what we got.

Reply:
> I don't know what your setup is on the Github side, but I would assume that "Xvfb" and friends are not installed.

The tests run on Travis and make heavy use of Xvfb, e.g.

https://travis-ci.org/tebeka/selenium/builds/266849658

The main test suite is apparently silently broken and I've opened #119:

https://travis-ci.org/tebeka/selenium/builds/373720652

Xvfb should still be available though.

> I'll switch to cmp.Diff. I usually use assert.Equal and friends, but I didn't want to mess with the dependencies of this package.

Thanks. I'm fine with adding dependencies especially for better testing. (Though I don't like the assert package in particular).